banjo/expression.cpp

```diff
@@ -37,4 +38,20 @@ make_requirements(Context& cxt,
   return cxt.make_requires(tparms, parms, reqs);
 }
 
+
+//TODO: Find a type for the Expr_list
+Expr&
+make_tuple_expr(Context& cxt, Expr_list& l)
+{
+  Type_list tlist;
+  for(auto eli = l.begin(); eli != l.end(); eli++) {
+    tlist.push_back(eli->type());
+  }
+  
+  Tuple_type t(tlist);
+  
+  return cxt.make_tuple_expr(t, l);
```

Comment:
How do I test for that segfault?
Are you referring to tlist, or t, or both?
Are you saying I shouldn't be using any local variables?
Thanks

On Wed, May 4, 2016 at 9:15 AM, Andrew Sutton <notifications@github.com>
wrote:

> In banjo/expression.cpp
> <https://github.com/asutton/banjo/pull/14#discussion_r62036058>:
>
> > @@ -37,4 +38,20 @@ make_requirements(Context& cxt,
> >    return cxt.make_requires(tparms, parms, reqs);
> >  }
> >
> > +
> > +//TODO: Find a type for the Expr_list
> > +Expr&
> > +make_tuple_expr(Context& cxt, Expr_list& l)
> > +{
> > +  Type_list tlist;
> > +  for(auto eli = l.begin(); eli != l.end(); eli++) {
> > +    tlist.push_back(eli->type());
> > +  }
> > +
> > +  Tuple_type t(tlist);
> > +
> > +  return cxt.make_tuple_expr(t, l);
>
> This will segfault. You are leaking a reference to a local variable.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/asutton/banjo/pull/14/files/abf73a25543526046288837c2f637be7edb9c535#r62036058>
>